src/pages/index.js

```diff
@@ -72,50 +88,166 @@ newCardPopup.setEventListeners();
 const popupImage = new PopupWithImage("#preview__modal");
 popupImage.setEventListeners();
 
+const avatarPopup = new PopupWithForm("#avatar__modal", handleAvatarUpdate);
+avatarPopup.setEventListeners();
+
+const deleteConfirm = new PopupWithConfirm("#delete__modal", handleCardDelete);
+deleteConfirm.setEventListeners();
+
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4792ec92-cf1c-45a6-8740-0f5d63585faa",
+    "Content-type": "application/json",
+  },
+});
+
 //////////////////////////////////////////// FUNCTIONS //////////////////////////////////////////////////////
 
 function handleImageClick(name, link) {
   popupImage.openPopup({ name, link });
 }
 
 function handleProfileModalSubmit(profileData) {
+  profileModalSubmitButton.textContent = "Saving...";
   console.log("profileData:", profileData);
   const name = profileData.title; //set name var to work with userInfo class
   const job = profileData.description; //set job var to work with userInfo class
   console.log("name:", name, "job:", job); //debug log
   user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
   const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
   console.log(updatedUserInfo); // debug log
   profilePopup.closePopup(); //close popup
+
+  setTimeout(() => {
+    profileModalSubmitButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

Reply:
* And this one hasn't been corrected https://github.com/RealnNTrue16/se_project_aroundtheus/pull/3/files#r1697656898